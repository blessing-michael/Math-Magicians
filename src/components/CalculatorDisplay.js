import React from 'react';

const CalculatorDisplay = () => (
  <>
    <h2 className="intro-text">Your Perfect Math Calculator</h2>
    <input type="text" placeholder={0} disabled />
  </>

);

export default CalculatorDisplay;
