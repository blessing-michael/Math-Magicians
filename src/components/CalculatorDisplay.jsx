import React from 'react';
import PropTypes from 'prop-types';

const CalculatorDisplay = ({ displayValue }) => (
  <>
    <h2 className="intro-text">Your Perfect Math Calculator</h2>
    <input type="text" value={displayValue} style={{ color: 'whitesmoke' }} disabled />
  </>
);

CalculatorDisplay.propTypes = {
  displayValue: PropTypes.string.isRequired,
};

export default CalculatorDisplay;
