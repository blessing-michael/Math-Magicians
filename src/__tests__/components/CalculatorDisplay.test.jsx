import React from 'react';
import renderer from 'react-test-renderer';
import CalculatorDisplay from '../../components/CalculatorDisplay';

test('Display of the Calculator renders correctly', () => {
  const tree = renderer.create(<CalculatorDisplay displayValue="100" />).toJSON();
  expect(tree).toMatchSnapshot();
});
