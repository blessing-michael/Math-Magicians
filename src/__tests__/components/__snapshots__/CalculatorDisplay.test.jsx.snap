// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display of the Calculator renders correctly 1`] = `
Array [
  <h2
    className="intro-text"
  >
    Your Perfect Math Calculator
  </h2>,
  <input
    disabled={true}
    style={
      Object {
        "color": "whitesmoke",
      }
    }
    type="text"
    value="100"
  />,
]
`;
